src/createSaleorTheme/createSaleorTheme.ts

```diff
@@ -0,0 +1,262 @@
+          '&$selected, &$selected:focus, &$selected:hover': {
+            backgroundColor: [colors.background.default, '!important'] as any,
+            color: colors.primary,
+            fontWeight: 700,
+          },
+          '&:hover': {
+            backgroundColor: [colors.background.default, '!important'] as any,
+            color: colors.font.default,
+            fontWeight: 400,
+          },
+          borderRadius: 4,
+        },
+      },
+      MuiSelect: {
+        root: {
+          '&$disabled': {
+            backgroundColor: colors.input.disabledBackground,
+          },
+        },
+      },
+      MuiSnackbarContent: {
+        action: {
+          '& $MuiIconButton': {
+            '& svg': {
+              color: colors.font.default,
+            },
+          },
+          display: 'block',
+          paddingBottom: 10,
+          paddingLeft: 0,
+          paddingRight: 45,
+        },
+        message: {
+          fontSize: 16,
+        },
+        root: {
+          backgroundColor: colors.background.paper,
+          boxShadow:
+            '0 6px 10px 0px rgba(0, 0, 0, 0.15), 0 1px 18px 0px rgba(0, 0, 0, 0.12), 0 3px 5px -1px rgba(0, 0, 0, 0.10)',
+          color: colors.font.default,
+          display: 'block',
+          maxWidth: 480,
+        },
+      },
+      MuiTouchRipple: {
+        child: {
+          backgroundColor: fade(colors.primary, 0.2),
+        },
+        childLeaving: {
+          backgroundColor: fade(colors.primary, 0.2),
+        },
+        ripple: {
+          '&$rippleVisible': {
+            backgroundColor: fade(colors.primary, 0.2),
+          },
+          borderRadius: '100%',
+        },
+      },
+    },
+    palette: createPalette(colors),
+    props: {
+      MuiFormControl: {
+        variant: 'filled',
+      },
+      MuiTextField: {
+        variant: 'outlined',
+      },
+      MuiCard: {
+        elevation: 0,
+      },
+      MuiTypography: {
+        component: 'div',
+      } as TypographyProps,
+      MuiCheckbox: {
+        color: 'primary',
+      },
+    },
+    shadows,
+    spacing: (value: number = 1) => `${(value * 8) / 10}rem`,
+    typography: {
+      allVariants: {
+        fontFamily,
+      },
+      body1: {
+        color: colors.font.default,
+        fontSize: '1.6rem',
+      },
+      body2: {
+        fontSize: '1.4rem',
+      },
+      button: {
+        fontSize: '1.4rem',
+      },
+      caption: {
+        fontSize: '1.2rem',
+      },
+      fontFamily,
+      h1: {
+        fontSize: '4.8rem',
+        fontWeight: 700,
+      },
+      h4: {
+        fontSize: '3.4rem',
+        color: colors.font.default,
+      },
+      h5: {
+        fontSize: '2.1rem',
+        fontWeight: 500,
+      },
+      h6: {
+        fontSize: '2rem',
+        fontWeight: 500,
+      },
+      subtitle1: {
+        fontSize: '1.6rem',
+      },
+      subtitle2: {
+        fontSize: '1.4rem',
+      },
+    },
+  }) as unknown) as SaleorTheme;
```

Comment:
What's wrong with the typing here?

Reply:
According to MUI typing: `theme.spacing(value: number): number` which isn't true, because we return string here.